src/core/tests/type_prop/non_max_suppression.cpp

```diff
@@ -369,6 +369,29 @@ TYPED_TEST_P(NMSDynamicOutputTest, interval_shapes_labels) {
     EXPECT_THAT(get_shape_labels(op->get_output_partial_shape(2)), Each(no_label));
 }
 
+TYPED_TEST_P(NMSDynamicOutputTest, num_box_dynamic_dim_max_boxes_per_class_as_const) {
+    auto boxes_shape = PartialShape{2, -1, 4};
+    auto scores_shape = PartialShape{2, {0, 5}, {1, 7}};
+    set_shape_labels(boxes_shape, 10);
+    set_shape_labels(scores_shape, 20);
+
+    const auto boxes = make_shared<op::v0::Parameter>(element::f32, boxes_shape);
+    const auto scores = make_shared<op::v0::Parameter>(element::f32, scores_shape);
+    const auto max_output_boxes_per_class = op::v0::Constant::create(element::i16, Shape{}, {5});
+    const auto iou_threshold = make_shared<op::v0::Parameter>(element::f32, Shape{});
+    const auto score_threshold = make_shared<op::v0::Parameter>(element::f32, Shape{});
+
+    const auto op = this->make_op(boxes, scores, max_output_boxes_per_class, iou_threshold, score_threshold);
+
+    EXPECT_THAT(op->outputs(),
+                ElementsAre(Property("Indicies shape", &Output<Node>::get_partial_shape, PartialShape({-1, 3})),
+                            Property("Scores shape", &Output<Node>::get_partial_shape, PartialShape({-1, 3})),
```

Comment:
Following my previous suggestion, in this test case, the expected `selected_boxes` upper bound could be set to `(max_output_boxes_per_class * num_batches * num_classes_upper_bound)` so `(5 * 2 * 5) = 50`. 
```suggestion
                ElementsAre(Property("Indicies shape", &Output<Node>::get_partial_shape, PartialShape({{0, 50}, 3})),
                            Property("Scores shape", &Output<Node>::get_partial_shape, PartialShape({{0, 50}, 3})),
```

Reply:
This behavior is to be compatible with previous implementation and selected boxes will calculated if  `num_boxes` has upper bound.

Reply:
Additional case when the bounds are set can be beneficial, but I approve current changes without this potential improvement.